File: codegen/results/v002/qwen3.5_4b_think/20260412_075620/python/002_word_frequency/solution.py
#!/usr/bin/env python3
import string
import re

# Read the file
with open('input/text.txt', 'r') as f:
    text = f.read()

# Tokenize the text
# Extract words (only alphabetic characters)
words = re.findall(r'[a-zA-Z]+', text.lower())

# Count word frequencies
word_count = {}
for word in words:
    word_count[word] = word_count.get(word, 0) + 1

# Sort by count descending, then alphabetically ascending for ties
sorted_words = sorted(word_count.items(), key=lambda x: (-x[1], x[0]))

# Output results
for word, count in sorted_words:
    print(f"{word}: {count}")

File: codegen/results/v002/qwen3.5_4b_think/20260412_075620/typescript/002_word_frequency/solution.ts
import { readFileSync, writeFileSync } from 'fs';

// Read the input file
const text = readFileSync('input/text.txt', 'utf8');

// Process text: convert to lowercase and extract words
const words = text
  .toLowerCase()
  .replace(/[^a-z0-9]/g, '') // Remove all characters that are not letters
  .split(/\s+/); // Split by whitespace

// Count frequency of each word
const frequency: Record<string, number> = {};
for (const word of words) {
  if (word) {
    frequency[word] = (frequency[word] || 0) + 1;
  }
}

// Sort by count descending, then alphabetically ascending
const sortedWords = Object.entries(frequency)
  .sort((a, b) => {
    if (b[1] !== a[1]) {
      return b[1] - a[1]; // Sort by count descending
    }
    return a[0].localeCompare(b[0]); // Sort alphabetically ascending for ties
  });

// Output results
for (const [word, count] of sortedWords) {
  console.log(`${word}: ${count}`);
}